File: 1021.lua
v,n,m=io.read("l"):gsub("[.]",""),{100,50,20,10,5,2},{100,50,25,10,5,1}
print("NOTAS:")
for _,i in pairs(n) do
	print(string.format("%d nota(s) de R$ %d.00",v//(i*100),i))
	v=v%(i*100)
end
print("MOEDAS:")
for _,i in pairs(m) do
	print(string.format("%d moeda(s) de R$ %.2f",v//i,i/100))
	v=v%i
end

File: cpp/1021.cpp
#include<iostream>
#include<iomanip>
#include<cmath>
using namespace std;

int main(){
    int i;
    double a, b[6] = {100,50,20,10,5,2}, c[6] = {100,50,25,10,5,1};
    cin >> a;
    a = a * 100;
    cout << "NOTAS: " << endl;
    for(i=0;i<6;i++){
        cout << floor(a / b[i] / 100) << " nota(s) de R$ " << b[i] << ".00" << endl;
        a = fmod(a, b[i] * 100);
    }
    cout << "MOEDAS: " << fixed << endl;
    for(i=0;i<6;i++){
        cout << setprecision(0) << floor(a / c[i]) << " moeda(s) de R$ " << setprecision(2) << c[i] / 100 << endl;
        a = fmod(a, c[i]);
    }
    return 0;
}